crates/swc_css_parser/src/parser/value/mod.rs

```diff
@@ -637,7 +637,43 @@ where
         let span = self.input.cur_span()?;
 
         if !is!(self, Dimension) {
-            return Err(Error::new(span, ErrorKind::Expected("Dimension")));
+            return Err(Error::new(span, ErrorKind::Expected("dimension token")));
+        }
+
+        match cur!(self) {
+            Token::Dimension { unit, .. } => {
+                match &*unit.to_ascii_lowercase() {
```

Comment:
This code seems bit problematic, as we don't use value of `&*unit.to_ascii_lowercase()`.